tests/rust/src/bin/readlink.rs

```diff
@@ -20,13 +20,6 @@ unsafe fn test_readlink(dir_fd: wasi::Fd) {
         "the remaining bytes should be untouched"
     );
 
-    // Read link into smaller buffer than the actual link's length
-    let buf = &mut [0u8; 4];
-    let err = wasi::path_readlink(dir_fd, "symlink", buf.as_mut_ptr(), buf.len())
-        .err()
-        .expect("readlink with too-small buffer should fail");
-    assert_errno!(err, wasi::ERRNO_RANGE);
```

Comment:
On [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/functions/readlink.html) and the current WAMR implementation, this doesn't cause an error - the buffer is filled with first `buflen` bytes from the link.

Reply:
Could we open a PR for the doc to clarify that then?

Reply:
Sure, https://github.com/WebAssembly/WASI/pull/554. I've also amended the test to enforce the POSIX/WAMR behaviour since it's more in keeping with docs; I would have thought that if we mention a POSIX function, it's safe to assume the behaviour is the same _unless_ specified otherwise.